lib/cadet/assessments/assessments.ex

```diff
@@ -612,6 +618,11 @@ defmodule Cadet.Assessments do
       |> join(:inner, [a, ..., s], st in assoc(s, :student))
       |> preload([_, q, g, s, st], question: q, grader: g, submission: {s, student: st})
 
+    if role in @grading_roles or role in @see_all_submissions_roles do
+      # ChatKit - create chatrooms if they don't exist
+      create_rooms(Submission |> where(id: ^id) |> Repo.one())
```

Comment:
Why are you creating rooms in this function. This function is called when graders at the frontend views a submission.

Reply:
I have answered this below, and have updated this PR's description with the current implementation.

> As we are now depending on an external service to host tutor comments, rooms are currently being created at three points - when answers are created, when students edit their answers and when cadets access answer in the grading view. In the event Chatkit crashes, the last two points serve as failsafes to ensure comments get to the students.

Reply:
I don't like this method. Can we try either creating a `mix task` that queries all `answer` in the database that is submitted but doesn't have a room, and generate the room key for them. Or we can try scheduling a task every 4 hours or every midnight to do the same. This way, we only need to generate the room key where it is really relevant to do so.

Reply:
oh i see you answered below